docs/devGuide/development/migratingToTypeScript.md

```diff
@@ -149,13 +174,12 @@ Combining the two steps into one - that is, adapting files immediately after ren
 
 The solution to this is to make the rename and adapt commits separate, in order for Git to recognize first that the file has changed into `.ts`, and so the changes are compared not against the old `.js` file, but to the renamed `.ts` file instead.
 
-### Why do we need a normal merge commit
+### Why do we need a rebase and merge commit
 
 Even if the migration developer has kept the history intact through the separate "Rename" and "Adapt" commits, this is only intact at their working branch. At the end of the day, the totality of the changes in the working branch is compared against the target branch, in which the same consideration would apply.
 
-If we do the usual squash commit, the changes from the two commits are combined into a new commit and only that commit will be pushed into the target branch. The original two commits are omitted, therefore the history of the working branch that we have tried to keep intact is stripped away.
-
-The normal merge commit also creates another commit (the merge commit), but the merge process interleaves the commits from the working branch to the target branch, therefore retaining the change history.
+If we do the usual squash commit, the changes from the two commits are combined into a new commit and only that commit will be pushed into the target branch. The original two commits are omitted, therefore the history of the working branch that we have tried to keep intact is stripped away. This results in a linear history, as if the changes in the working branch were made
```

Comment:
I don't think the point about linear history is necessarily relevant here - our repo history is quite linear because of the squash and merge strategy so using a rebase over squash+merge doesn't add much linearity. Maybe can we revert this change?

Reply:
@ong6 this is not reverted yet?